packages/desktop-client/src/components/mobile/transactions/TransactionEdit.jsx

```diff
@@ -463,30 +422,22 @@ const TransactionEditInner = memo(function TransactionEditInner({
   const { grouped: categoryGroups } = useCategories();
 
   const [transaction, ...childTransactions] = transactions;
+  const transactionsRef = useRef(transactions);
+  useEffect(() => {
+    transactionsRef.current = transactions;
+  }, [transactions]);
 
-  const { editingField, onRequestActiveEdit, onClearActiveEdit } =
-    useSingleActiveEditForm();
-  const [totalAmountFocused, setTotalAmountFocused] = useState(true);
   const childTransactionElementRefMap = useRef({});
   const hasAccountChanged = useRef(false);
 
   const payeesById = useMemo(() => groupById(payees), [payees]);
   const accountsById = useMemo(() => groupById(accounts), [accounts]);
 
-  const onTotalAmountEdit = useCallback(() => {
-    onRequestActiveEdit?.(getFieldName(transaction.id, 'amount'), () => {
-      setTotalAmountFocused(true);
-      return () => setTotalAmountFocused(false);
-    });
-  }, [onRequestActiveEdit, transaction.id]);
-
-  const isInitialMount = useInitialMount();
-
-  useEffect(() => {
-    if (isInitialMount && adding) {
-      onTotalAmountEdit();
-    }
-  }, [adding, isInitialMount, onTotalAmountEdit]);
+  // getTransaction prevents stale-closure issue with dialogs
+  const getTransaction = useCallback(
```

Comment:
What scenario are you encountering an issue with this?

Reply:
When clicking on a next editable field away from field which is currently being edited, currently edited field is not being saved, because older version of transaction is baked in modal `onSelect` callback (state is stale in `onSelect` closure). 

Reproducible example: 
1. Edit "Amount" from 0.00 to 1.00
2. Click on "Payee" select field
3. In a dialog choose any value
Observed: Amount is being reset back to 0.00

What happens is when "Amount" field loses focus it fires `onUpdate` event to update transaction amount from 0.00 to 1.00. At the same time "Payee" field react on click and opens a modal, and yet unmodified transaction (with amount 0.00) is captured in modal `onSelect` callback. So when the a value is selected in modal, it fires `onUpdate` event to update Payee, but with a previous amount value. 
 
 ```typescript
pushModal('payee-autocomplete', {
    onSelect: payeeId => {
        onUpdateInner(transactionToEdit, name, payeeId);
        //                        ^----- unmodified transaction captured
    },
}),
 ```